tesseract_ros/tesseract_monitoring/src/environment_monitor.cpp

```diff
@@ -110,6 +110,33 @@ EnvironmentMonitor::~EnvironmentMonitor()
   shutdown();
 }
 
+bool EnvironmentMonitor::waitForConnection(ros::Duration timeout) const
+{
+  const ros::WallTime start_time = ros::WallTime::now();
+  const ros::WallDuration wall_timeout{ timeout.toSec() };
+  bool is_connected{ false };
+  while (ros::ok())
+  {
+    {
+      auto lock = std::shared_lock(scene_update_mutex_);
+      is_connected = tesseract_->isInitialized();
+    }
+    if (is_connected)
+      return true;
+
+    if (wall_timeout >= ros::WallDuration(0))
```

Comment:
Should this be > only to allow disabling timeout?

Reply:
I took a look at the waitForService code and this is what they did. The default value is `ros::Duration(-1)`, but I assume someone could pass in `ros::Duration(0)` which would indicate only to run the check once and then return. If it was greater than only it would wait indefinitely.